### PR TITLE
Use the crowbarctl upgrade command for services and nodes actions

### PR DESCRIPTION
**Pull request by jsuchome**

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -4781,18 +4782,27 @@ function onadmin_crowbar_nodeupgrade
         fi
 
         if [[ $want_nodesupgrade ]]; then
-            # FIXME: crowbarctl command can time out easily
-            # Do not use it until https://bugzilla.novell.com/show_bug.cgi?id=997293 is fixed
-            # safely crowbarctl upgrade services
-            curl -s -X POST -H "$crowbar_api_v2_header" $crowbar_api/api/upgrade/services
+            safely crowbarctl upgrade services
+            wait_for 10 1 "grep -q running $upgrade_progress_file" "services step to start"
+            wait_for 300 5 "! grep -q running $upgrade_progress_file" "services step to finish"
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: this way of tracking state is prone to races - e.g. if an upgrade finishes too fast (e.g. under 1 second), you will not see it and error out in the first wait.

Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Hm, then we might add a check for error right after the first running check...

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -69,6 +69,7 @@ crowbar_install_log=/var/log/crowbar/install.log
 crowbar_init_api=http://localhost:4567/api
 crowbar_lib_dir=/var/lib/crowbar
 crowbar_api_v2_header="Accept: application/vnd.crowbar.v2.0+json"
+upgrade_progress_file=/var/lib/crowbar/upgrade/6-to-7-progress.yml
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: maybe we can wrap this definition in a cloudsource/upgrade_cloudsource check

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -4781,18 +4782,26 @@ function onadmin_crowbar_nodeupgrade
...
                 crowbarctl upgrade status
-                complain 12 "Services step has failed. Check the upgrade status."
+                complain 12 "'Services' step has failed. Check the upgrade status."
             fi
 
-            # safely crowbarctl upgrade nodes
-            curl -s -X POST -H "$crowbar_api_v2_header" $crowbar_api/api/upgrade/nodes
+            # This is just a fake command now for OpenStack DB backup
+            # FIXME: replace with crowbarctl upgrade once available
+            curl -s -X POST -H "$crowbar_api_v2_header" $crowbar_api/api/upgrade/openstackbackup
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: it's available now, and should be in the product by now. see https://github.com/crowbar/crowbar-client/releases/tag/v3.1.6

Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Thanks

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Where exactly would you do that?

Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: i thought around the definition, something like:

```bash
if [[ "$cloudsource" == "develcloud6" ]] && [[ "$upgrade_cloudsource" == "develcloud7" ]]
    upgrade_progress_file=/var/lib/crowbar/upgrade/6-to-7-progress.yml
fi
```

... btw this code is not tested

Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Maybe, yes. But maybe we could add this once we have different possible values for the progress file

**Comment by jsuchome** [comment]
@bmwiedemann do you think it's usable this way?

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -4781,18 +4782,26 @@ function onadmin_crowbar_nodeupgrade
...
+            if grep -q "failed" $upgrade_progress_file ; then
                 crowbarctl upgrade status
-                complain 12 "Services step has failed. Check the upgrade status."
+                complain 12 "'Services' step has failed. Check the upgrade status."
             fi
 
-            # safely crowbarctl upgrade nodes
-            curl -s -X POST -H "$crowbar_api_v2_header" $crowbar_api/api/upgrade/nodes
+            # This is just a fake command now for OpenStack DB backup
+            # FIXME: add a proper check for failure once we have a real thing
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: what does 'a real thing' refer to here? could be clarified.

Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Yes, it does. I've adapted the comment

**Pull request merged**